USB: netusb: Move status callback to functions

finikorg opened the pull request:
Fix #7364

finikorg committed:
usb: netusb: Add subsys/usb to include list

This helps to source files without ../../ etc paths.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
usb: netusb: Use function - defined status callbacks

Allow functions to define status callbacks

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
usb: netusb: Implement status callback for ECM

Implement status callback.

Fixes: #7364

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
usb: netusb: Implement status callback for EEM

Implement status callback.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
usb: netusb: Implement status callback for RNDIS

Implement status callback

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

jfischer-phytec-iot commented on subsys/usb/class/netusb/function_rndis.c:
```diff
@@ -1060,10 +1060,48 @@ static struct usb_ep_cfg_data rndis_ep_data[] = {
+		break;
+	case USB_DC_CONFIGURED:
+		SYS_LOG_DBG("USB device configured");
+		netusb_enable();
```
That was necessary here because of detection on the Windows OS host, right?

finikorg committed:
samples: echo_server: Remove conflicting option from conf file

Remove conflicting option CONFIG_NET_IPV6 since it got selected due to
CONFIG_NET_APP_NEED_IPV6 in netusb configuration file.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

jfischer-phytec-iot commented on subsys/usb/class/netusb/function_ecm.c:
```diff
@@ -197,10 +198,60 @@ static struct usb_ep_cfg_data ecm_ep_data[] = {
+		SYS_LOG_DBG("USB device configured");
+		break;
+	case USB_DC_DISCONNECTED:
+		SYS_LOG_DBG("USB device disconnected");
```
@finikorg What do you think about removing unused cases, so that e.g. only USB_DC_DISCONNECTED and USB_DC_INTERFACE remain?

finikorg committed:
usb: netusb: Refactor function's status callbacks

Refactor functions status callbacks for netusb ECM, EEM and RNDIS.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>
